src/components/AutocompleteInput/AutocompleteInput.tsx

```diff
@@ -14,25 +14,30 @@ type Option = PropsWithSpread<
   HTMLProps<HTMLOptionElement>
 >;
 
-type Props = PropsWithSpread<
-  {
-    options?: (string | Option)[] | null;
-  },
-  InputProps
->;
+const isOption = (option: any): option is Option =>
+  option instanceof Object &&
+  option?.label &&
+  typeof option?.value === "string";
+
+type Props = PropsWithSpread<{ options: unknown[] }, InputProps>;
 
 const generateOptions = (options: Props["options"]) =>
-  options?.map((option) => {
-    if (typeof option === "string") {
-      return <option value={option} key={option} />;
-    }
-    const { value, label, ...props } = option;
-    return (
-      <option value={value} key={value} {...props}>
-        {label}
-      </option>
-    );
-  });
+  options
+    .filter((option) => typeof option === "string" || isOption(option))
+    .map((option) => {
+      if (typeof option === "string") {
+        return <option value={option} key={option} />;
+      }
+      if (isOption(option)) {
+        const { value, label, ...props } = option;
+        return (
+          <option value={value} key={value} {...props}>
+            {label}
+          </option>
+        );
+      }
+      return null;
```

Comment:
This is theoretically redundant (considering we are using the `filter` method before it), but I think `map` method expects to get a return for all potential paths.

Reply:
Codecov also fails, cause I can't test this line as the code will never arrive here.